### PR TITLE
feat(CO-49) Frontend Dungeon Enemy List

### PR DESCRIPTION
**Pull request by matyaspeterfi**

### CODE REVIEW

File: src/app/components/enemy-list/enemy-list.component.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<div class='container'>
+  <div class="enemyList" *ngIf="dungeon">
+    Obstacles:
+    <div class='row' *ngFor="let item of dungeon.obstacles; let i = index" [attr.data-index]="i" (click)="selectObstacle(item)">
```
Comment (juhaszPeterDev) [inline src/app/components/enemy-list/enemy-list.component.html]: solve this without using indexes
use encapsulation principles
test for changing values

Comment (matyaspeterfi) [reply src/app/components/enemy-list/enemy-list.component.html]: fixed, thanks!

File: src/app/components/enemy-list/enemy-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  selector: 'app-enemy-list',
+  templateUrl: './enemy-list.component.html',
+  styleUrls: ['./enemy-list.component.scss']
+})
+export class EnemyListComponent implements OnChanges {
+  _selectedObstacle: Obstacle;
+
+  @Input() dungeon: Dungeon;
+  @Output() _valueChange = new EventEmitter;
+  deadObstacles: string[] = [];
```
Comment (juhaszPeterDev) [inline src/app/components/enemy-list/enemy-list.component.ts]: this seems to be unnecessary

File: src/app/components/enemy-list/enemy-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.dungeon = changes.dungeon.currentValue;
```
Comment (juhaszPeterDev) [inline src/app/components/enemy-list/enemy-list.component.ts]: this seems to be unnecessary

Comment (matyaspeterfi) [reply src/app/components/enemy-list/enemy-list.component.ts]: fixed, thanks!

**Pull request merged**